File: python/routers/create_database.py
""" 
This file is part of the Nervatura Framework
http://nervatura.com
Copyright © 2011-2023, Csaba Kappel
License: LGPLv3
https://raw.githubusercontent.com/nervatura/nervatura/master/LICENSE 
"""

import os
from fastapi import APIRouter, Request, Form
from fastapi.responses import HTMLResponse
from fastapi.templating import Jinja2Templates

from lib.api import getApi

router = APIRouter(prefix="/create_database")
templates = Jinja2Templates(directory="templates")

login_data = {
  "api_key": os.getenv("NT_API_KEY"),
  "alias": "demo",
  "demo": "true",
  "api_type": os.getenv("NT_EXAMPLE_DEFAULT_API"),
  "title": "Create a demo database",
  "error": False, "result": False,
  "env": {
    "NT_API_KEY": os.getenv("NT_API_KEY"),
    "NT_ALIAS_DEMO": os.getenv("NT_ALIAS_DEMO")
  }
}

@router.get("/", response_class=HTMLResponse)
async def index(request: Request):
  data = login_data | { "request": request }
  return templates.TemplateResponse("create_database.html", data)

@router.post("/", response_class=HTMLResponse)
async def login(
  request: Request,
  api_key: str = Form(...),
  alias: str = Form(...),
  demo: str = Form(...),
  api_type: str = Form(...),
):
  result = getApi(api_type, "DatabaseCreate")(
    api_key, 
    { "database": alias, "demo": True if demo == "true" else False })
  data = login_data | { "api_key": api_key,
    "alias": alias, "demo": demo, "api_type": api_type,
    "error": result[1] if result[1] is not None else False,
    "result": result[0] if result[1] is None else False,
    "request": request
  }
  return templates.TemplateResponse("create_database.html", data)

File: php/routes/CreateDatabase.php
<?php
/*
This file is part of the Nervatura Framework
http://nervatura.com
Copyright © 2011-2023, Csaba Kappel
License: LGPLv3
https://raw.githubusercontent.com/nervatura/nervatura/master/LICENSE
*/

class CreateDatabase {
  function init($f3){
    $f3->set('title', 'Create a demo database');
    $f3->set('alias', 'demo');
    $f3->set('demo', 'true');
    $f3->set('error', NULL);
    $f3->set('result', NULL);
    $f3->set('api_key', getenv('NT_API_KEY'));
    $f3->set('api_type', getenv('NT_EXAMPLE_DEFAULT_API'));
    $f3->set('env', array(
      'NT_API_KEY'=>getenv('NT_API_KEY'),
      'NT_ALIAS_DEMO'=>getenv('NT_ALIAS_DEMO')
    ));
  }
  
  public function get($f3){
    $this->init($f3);
    $this->sendResult($f3);
  }

  public function post($f3){
    $this->init($f3);
    $f3->set('alias', $f3->get('POST.alias'));
    $f3->set('demo', $f3->get('POST.demo'));
    $f3->set('api_key', $f3->get('POST.api_key'));
    $f3->set('api_type', $f3->get('POST.api_type'));
    $api = $f3->get($f3->get('api_type'));
    $options = array(
      'demo' => $f3->get('demo'),
      'alias' => $f3->get('alias')
    );
    $result = $api->DatabaseCreate($f3->get('api_key'), $options);
    $f3->set('error', $result['error']);
    $f3->set('result', $result['result']);
    $this->sendResult($f3);
  }

  function sendResult($f3){
    $f3->set('content', 'create_database.html');
    echo Template::instance()->render('layout.html');
  }
}

?>